src/Data/LLVM/BitCode/IR/Function.hs

```diff
@@ -122,15 +122,18 @@ type DeclareList = Seq.Seq FunProto
 -- | Turn a function prototype into a declaration.
 finalizeDeclare :: FunProto -> Parse Declare
 finalizeDeclare fp = case protoType fp of
-  PtrTo (FunTy ret args va) -> return Declare
+  PtrTo adr (FunTy ret args va) -> return Declare
     { decLinkage    = protoLinkage fp
     , decVisibility = protoVisibility fp
+    , decUnnamedAddr = protoUnnamedAddr fp
     , decRetType    = ret
     , decName       = protoSym fp
     , decArgs       = args
     , decVarArgs    = va
     , decAttrs      = []
     , decComdat     = protoComdat fp
+    -- TODO what if this isn't equal to adr?
+    , decAddrSpace  = protoAddrSpace fp
```

Comment:
I think this would still be a sensible construct if they were not equal, but it hasn't been used.

Reply:
Hm, that is tricky. Do you know what LLVM does here? If nothing else, it might be worth checking for an (opt-in) `assert`.